src/rmview/connection.py

```diff
@@ -62,56 +62,88 @@ class rMConnect(QRunnable):
       self.signals.onConnect.connect(onConnect)
     if callable(onError):
       self.signals.onError.connect(onError)
 
+  def _initialize(self):
+    # NOTE: Loading system known hosts can take a long time that's why it should happen inside
+    # run() so it doesn't block the main qt render loop which will cause main QT window to freeze
+    # until the loading completes.
     try:
       self.client = paramiko.SSHClient()
 
-      if host_key_policy != "ignore_all":
-        self.client.load_system_host_keys()
-        self._known_hosts = known_hosts
-        if known_hosts and os.path.isfile(known_hosts):
-          self.client.load_system_host_keys(known_hosts)
-          self.client.load_host_keys(known_hosts) # to preserve them when saving
-          log.info("LOADED %s", known_hosts)
-
-      policy = HOST_KEY_POLICY.get(host_key_policy, RejectNewHostKey)
+      if self.host_key_policy != "ignore_all":
+        if self.known_hosts and os.path.isfile(self.known_hosts):
+          log.info("Using known hosts file: %s" % (self.known_hosts))
+          self.client.load_host_keys(self.known_hosts)
+          log.info("Loaded known hosts from %s", self.known_hosts)
+        else:
+          log.info("Using system default known hosts file")
+          log.info("Loading system default known hosts file, this may take a while...")
+          # ideally we would want to always load the system ones
+          # and have the local keys have precedence, but paramiko gives
+          # always precedence to system keys
+          # There is extremly slow in system with many known host entries... :/
+          # See https://github.com/paramiko/paramiko/issues/191
+          self.client.load_system_host_keys()
+          log.info("System default known host file loaded")
+
+      policy = HOST_KEY_POLICY.get(self.host_key_policy, RejectNewHostKey)
       self.client.set_missing_host_key_policy(policy())
 
-      if key is not None:
-        key = os.path.expanduser(key)
-        try:
-          pkey = paramiko.RSAKey.from_private_key_file(key)
-        except paramiko.ssh_exception.PasswordRequiredException:
-          passphrase, ok = QInputDialog.getText(None, "Configuration","SSH key passphrase:", QLineEdit.Password)
-          if ok:
-            pkey = paramiko.RSAKey.from_private_key_file(key, password=passphrase)
-          else:
-            raise Exception("A passphrase for SSH key is required")
+      if self.key is not None:
+        key = os.path.expanduser(self.key)
+
+        if self.password:
+            # password protected key file, password provided in the config
+            pkey = paramiko.RSAKey.from_private_key_file(key, password=self.password)
+        else:
+            try:
+                pkey = paramiko.RSAKey.from_private_key_file(key)
+            except paramiko.ssh_exception.PasswordRequiredException:
+                passphrase, ok = QInputDialog.getText(None, "Configuration","SSH key passphrase:",
```

Comment:
This cannot be invoked in the connection thread! Should be moved back to `__init__`

Reply:
I forget the details now, but ``_initialize()`` which loads host keys also can't be called inside the main thread since it will block the main qt draw loop which will freeze the whole app / window until the host keys are loaded.

So likely need to move it in a background thread or similar.